src/api.rs

```diff
@@ -616,6 +639,14 @@ impl Api {
         let resp = self
             .request(Method::Post, &path)?
             .with_form_data(form)?
+            .with_retry(
+                self.config.get_max_retry_count().unwrap(),
+                &[
+                    HTTP_STATUS_502_BAD_GATEWAY,
+                    HTTP_STATUS_503_SERVICE_UNAVAILABLE,
+                    HTTP_STATUS_504_GATEWAY_TIMEOUT,
+                ],
+            )?
```

Comment:
Just a note here, the file uploads are not really idempotent ("File already present" might be returned on a retry).